test/gtest/uct/test_mem.cc

```diff
@@ -106,25 +107,54 @@ UCS_TEST_P(test_mem, md_alloc) {
         status = uct_md_query(md, &md_attr);
         ASSERT_UCS_OK(status);
 
-        for (nonblock = 0; nonblock <= 1; ++nonblock) {
-            int flags = nonblock ? UCT_MD_MEM_FLAG_NONBLOCK : 0;
+        for (mem_type = UCS_MEMORY_TYPE_HOST; mem_type < UCS_MEMORY_TYPE_LAST; mem_type++) {
+            params.mem_type = (ucs_memory_type_t)mem_type;
+            if (md_attr.cap.alloc_mem_types & UCS_BIT(mem_type)) {
+                if (mem_type == UCS_MEMORY_TYPE_HOST) {
+                    for (nonblock = 0; nonblock <= 1; ++nonblock) {
```

Comment:
most of the code in if/else is same. can you move this for loop above and set nonblock  flag only for HOST and nonblock=1.